transient_auth/sync_hashes.py

```diff
@@ -0,0 +1,117 @@
+import os
+import tempfile
+import uuid
+from git import Repo
+from datetime import datetime
+from selenium import webdriver
+from pathlib import Path
+from transient_auth.models import Commit, Hash, Edition, Repository
+from transient_auth.utils import calc_binary_content_hash, calc_page_hash
+
+options = webdriver.ChromeOptions()
+options.add_argument("headless")
+driver = webdriver.Chrome(chrome_options=options)
+
+EMPTY_TREE_SHA = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+
+def sync_hashes(repo_path):
+  repo_path = Path(repo_path)
+  repo_name = '{}/{}'.format(repo_path.parent.name, repo_path.name)
+  repo = Repo(str(repo_path))
+  repo_commits = list(repo.iter_commits('master'))[::-1]
+
+  # TODO we need to know when an edition ends and where it begins
+  # since editions are on branches, maybe use them
+  # or call this edition by edition
+
+  try:
+    repository = Repository.objects.get(name=repo_name)
+    edition = Edition.objects.filter(repository=repository).latest('id')
+  except Repository.DoesNotExist:
+    repository = Repository(name=repo_name)
+    repository.save()
+    edition = None
+  except Repository.MultipleObjectsReturned:
+    print('Multiple repositories of the same name found. That should not be the case.')
+    return
+
+  if edition is None:
+    # create the initial edition
+    commit_date = repo.git.show(s=True, format='%ci {}'.format(repo_commits[0].hexsha)).split()[0]
+    edition_name = commit_date.rsplit('-', 1)[0]
+    edition = Edition(name=edition_name, date=commit_date, repository=repository)
+    edition.save()
+
+  # check if commits are already in the database
+  # if they are, see if there are commits which have not been inserted yet
+  # if not, insert the hashes from the beginning
+  inserted_commits = Commit.objects.filter(edition=edition)[::1]
+  if len(inserted_commits) == len(repo_commits):
+    print('All commits have been loaded into the database')
+    return
+
+  # find the last inserted commit
+  inserted_commits_num = len(inserted_commits)
+  if inserted_commits_num == 0:
+    prev_commit = Commit(sha=EMPTY_TREE_SHA)
+  else:
+    prev_commit = inserted_commits[-1]
+
+  for commit in repo_commits[inserted_commits_num::]:
+    date = datetime.utcfromtimestamp(commit.committed_date).strftime('%Y-%m-%d %H:%M')
```

Comment:
date has not been authenticated, so cannot use.

Reply:
After giving it some thought, I think that using `through-date` stored in `metadata.json` is the best option.

Reply:
In case of `html` repositories, we could use `through-date` stored in `metadata.json`. I don't think we have the same information in case of pdf repositories. Should all repositories contain a metadata file?

Reply:
We decided to address this later, once we have all of the necessary information